qiskit/algorithms/time_evolvers/variational/var_qrte.py

```diff
@@ -0,0 +1,112 @@
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.primitives import BaseEstimator
+
+from .solvers.ode.forward_euler_solver import ForwardEulerSolver
+
+from .variational_principles import RealVariationalPrinciple, RealMcLachlanPrinciple
+from .var_qte import VarQTE
+
+from ..real_time_evolver import RealTimeEvolver
+
+
+class VarQRTE(VarQTE, RealTimeEvolver):
+    """Variational Quantum Real Time Evolution algorithm.
+
+    .. code-block::python
+
+        from qiskit.algorithms import TimeEvolutionProblem, VarQITE
+        from qiskit.circuit.library import EfficientSU2
+        from qiskit.algorithms.time_evolvers.variational import RealMcLachlanPrinciple
+        from qiskit.quantum_info import SparsePauliOp
+        import numpy as np
+
+        observable = SparsePauliOp.from_list(
+            [
+                ("II", 0.2252),
+                ("ZZ", 0.5716),
+                ("IZ", 0.3435),
+                ("ZI", -0.4347),
+                ("YY", 0.091),
+                ("XX", 0.091),
+            ]
+        )
+
+        ansatz = EfficientSU2(observable.num_qubits, reps=1)
+        init_param_values = np.zeros(len(ansatz.parameters))
+        for i in range(len(ansatz.parameters)):
+            init_param_values[i] = np.pi / 2
+        var_principle = RealMcLachlanPrinciple()
+        time = 1
+        evolution_problem = TimeEvolutionProblem(observable, time)
+        var_qrte = VarQRTE(ansatz, var_principle, init_param_values)
+        evolution_result = var_qite.evolve(evolution_problem)
+    """
+
+    def __init__(
+        self,
+        ansatz: QuantumCircuit,
+        initial_parameters: dict[Parameter, float] | list[float] | np.ndarray,
+        variational_principle: RealVariationalPrinciple | None = None,
+        estimator: BaseEstimator | None = None,
+        ode_solver: Type[OdeSolver] | str = ForwardEulerSolver,
+        lse_solver: Callable[[np.ndarray, np.ndarray], np.ndarray] | None = None,
+        num_timesteps: int | None = None,
+        imag_part_tol: float = 1e-7,
+        num_instability_tol: float = 1e-7,
+    ) -> None:
+        r"""
+        Args:
+            ansatz: Ansatz to be used for variational time evolution.
+            initial_parameters: Initial parameter values for an ansatz.
+            variational_principle: Variational Principle to be used. Defaults to
+                ``RealMcLachlanPrinciple``.
+            estimator: An estimator primitive used for calculating expectation values of
+                TimeEvolutionProblem.aux_operators.
+            ode_solver: ODE solver callable that implements a SciPy ``OdeSolver`` interface or a
+                string indicating a valid method offered by SciPy.
+            lse_solver: Linear system of equations solver callable. It accepts ``A`` and ``b`` to
+                solve ``Ax=b`` and returns ``x``. If ``None``, the default ``np.linalg.lstsq``
+                solver is used.
```

Comment:
> If ``None``, the default ``np.linalg.lstsq``solver is used.

I guess the comment I made about the default dependence runs deeper - I guess if we can anticipate that never changing then mentioning the default in many places we can live with,

Reply:
I will keep the default only where it's actually applied.